src/main/java/com/glovo/ttglovo/securityManagement/appuser/AppUser.java

```diff
@@ -0,0 +1,172 @@
+package com.glovo.ttglovo.securityManagement.appuser;
+
+import com.glovo.ttglovo.cart.CartItem;
+import lombok.*;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+
+import javax.persistence.*;
+import javax.validation.constraints.Email;
+import javax.validation.constraints.NotEmpty;
+import java.util.*;
+
+@Builder
+@Getter
+@Setter
+@AllArgsConstructor
+@NoArgsConstructor
+@Entity(name = "AppUser")
+@Table(name = "user_table",
+        uniqueConstraints = {
+                @UniqueConstraint(
+                        columnNames = "email",
+                        name = "uk_email"
+                )
+        }
+)
+public class AppUser implements UserDetails {
+
+    @Id
+    @SequenceGenerator(
+            name = "user_sequence",
+            sequenceName = "user_sequence",
+            allocationSize = 1
+    )
+    @GeneratedValue(
+            strategy = GenerationType.SEQUENCE,
+            generator = "user_sequence"
+    )
+    @Column(name = "id")
+    private Long id;
+
+    @NotEmpty(message = "First name may not be empty")
+    @Column(
+            name = "first_name",
+            nullable = false,
+            columnDefinition = "TEXT"
+    )
+    private String firstName;
+
+    @NotEmpty(message = "Last name may not be empty")
+    @Column(
+            name = "last_name",
+            nullable = false,
+            columnDefinition = "TEXT"
+    )
+    private String lastName;
+
+    @Email(regexp = ".+@.+\\..+")
+    @Column(
+            name = "email",
+            nullable = false,
+            columnDefinition = "TEXT"
+    )
+    private String email;
+
+    @NotEmpty(message = "Password may not be empty")
+    @Column(
+            name = "password",
+            nullable = false,
+            columnDefinition = "TEXT"
+    )
+    private String password;
+
+// no username, only first name
+
+    @Column(
+            name = "app_user_role",
+            nullable = false,
+            columnDefinition = "TEXT"
+    )
+    @Enumerated(EnumType.STRING)
+    private AppUserRole appUserRole;
+
+    private Boolean locked = false;
+
+    private Boolean enabled = false;
+
+    @OneToMany(
+            mappedBy = "user",
+            orphanRemoval = true,
+            cascade = {CascadeType.PERSIST, CascadeType.REMOVE},
+            fetch = FetchType.LAZY
+    )
+    private List<CartItem> cartItems = new ArrayList<>();
+
+    public AppUser(String firstName,
+                   String lastName,
+                   String email,
+                   String password,
+                   AppUserRole appUserRole) {
+        this.firstName = firstName;
+        this.lastName = lastName;
+        this.email = email;
+        this.password = password;
+        this.appUserRole = appUserRole;
+    }
```

Comment:
Since you're using @AllArgsConstructor this one is not needed (it will be automatically generated)